src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -167,41 +167,49 @@ def get_paths_to_platform_and_hwsku_dirs():
     # Get platform and hwsku
     (platform, hwsku) = get_platform_and_hwsku()
 
+    platform_path = None
+    hwsku_path = None
     # Determine whether we're running in a container or on the host
-    platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)
+    if platform:
+        platform_path_host = os.path.join(HOST_DEVICE_PATH, platform)
 
-    if os.path.isdir(CONTAINER_PLATFORM_PATH):
+    if  os.path.isdir(CONTAINER_PLATFORM_PATH):
         platform_path = CONTAINER_PLATFORM_PATH
-    elif os.path.isdir(platform_path_host):
+    elif platform_path_host and os.path.isdir(platform_path_host):
         platform_path = platform_path_host
     else:
         raise OSError("Failed to locate platform directory")
 
-    hwsku_path = os.path.join(platform_path, hwsku)
+    if hwsku:
+        hwsku_path = os.path.join(platform_path, hwsku)
```

Comment:
These changes should not be necessary. `platform` and `hwsku` should never be `None`.

The VS test failure should be fixed by this PR that I just opened: https://github.com/Azure/sonic-buildimage/pull/5266

Reply:
Thanks, 
I will revert this change, Just wanted to see if the Build passes along with these changes.

Reply:
As discussed offline, this scenario should never arise. Every platform, physical or virtual, should have a platform string and HwSKU. These functions should *never* return `None`, so this is not a potential fix.

Reply:
fixed reverted the change, build will fail for now, but it should be fixed once #5266 is merged